jetty-io/src/main/java/org/eclipse/jetty/io/ByteBufferPool.java

```diff
@@ -78,7 +72,17 @@ default ByteBuffer newByteBuffer(int capacity, boolean direct)
         return direct ? BufferUtil.allocateDirect(capacity) : BufferUtil.allocate(capacity);
     }
 
-    public static class Lease
+    /**
+     * Get this pool as a {@link RetainableByteBufferPool}, which supports reference counting of the
+     * buffers and possibly a more efficient lookup mechanism based on the {@link org.eclipse.jetty.util.Pool} class.
+     * @return This pool wrapped as a RetainableByteBufferPool.
+     */
+    default RetainableByteBufferPool asRetainableByteBufferPool()
```

Comment:
Perhaps.    We can return a new instance here because it is a non allocating one - ie it is the null implementations. So would it be OK to say that if the pool returned does actually retain buffers then the pool returned must always be the same (or a wrapper of the same) retained pool.

Reply:
removed default impl and javadoc now says must be the same instance always